libobs/obs-encoder.c

```diff
@@ -694,6 +694,25 @@ void obs_encoder_shutdown(obs_encoder_t *encoder)
 	pthread_mutex_unlock(&encoder->init_mutex);
 }
 
+void obs_encoder_get_video_info(struct obs_encoder *encoder,
+				struct video_scale_info *info)
+{
+	const struct video_output_info *voi;
+	voi = video_output_get_info(encoder->media);
+
+	info->format = voi->format;
+	info->colorspace = voi->colorspace;
+	info->range = voi->range;
+	info->width = obs_encoder_get_width(encoder);
+	info->height = obs_encoder_get_height(encoder);
+
+	if (encoder->info.get_video_info)
+		encoder->info.get_video_info(encoder->context.data, info);
+
+	if (info->width != voi->width || info->height != voi->height)
```

Comment:
What's happening here? I don't think it's a good idea to be changing any data with a `get` method.

Reply:
This is just a copy of the existing `get_video_info`, also a leftover from the original ARGB stuff and I think it could be removed as well.

https://github.com/obsproject/obs-studio/blob/fbf6c8d243714b5d5409826f69c633c627d3535c/libobs/obs-encoder.c#L168

Reply:
It seemed useful and I assumed it was used in the ARGB stuff, but I also cannot find it used in the PR (and the old get_video_info is still leftover). Given its not actually used Ill remove it and revert to the internal.

That said do we still need that set on this path?

Reply:
The video info feature is to allow the encoder to specify the input format it wants, in theory I suppose this could include a different size. I'd leave the existing internal function alone for now.